htmlmediaelement: Support seek requests for non seekable fetch context

tharkum opened the pull request:
According to specification the data:// URL protocol doesn't support range request
so be able make any seek request to required content position let's allow
for non seekable fetch context discard fetched content bytes until seek offset.
https://fetch.spec.whatwg.org/#scheme-fetch

Some scheme URLs (like data:// URL) doesn't expose "Content-Length" header in response
so the total expected size of the stream is unknown and it causes some additional
seek request (SeekData) from the media player. Try to post configure stream size
after we reached fetch EOS response.

Related source code which breaks WPT tests:
[tests/wpt/tests/html/canvas/element/manual/imagebitmap/common.sub.js#L56-L78](https://github.com/servo/servo/tree/main/tests/wpt/tests/html/canvas/element/manual/imagebitmap/common.sub.js#L56-L78)

Testing: Improvements in the following tests:
 - html/canvas/element/manual/imagebitmap/createImageBitmap*

Fixes: https://github.com/servo/servo/issues/32645
Fixes: https://github.com/servo/servo/issues/32745
Fixes: https://github.com/servo/servo/issues/34119
Fixes: https://github.com/servo/servo/issues/34120
Fixes: https://github.com/servo/servo/issues/34151

jdm reviewed (approved):
Nice solution! Let's merge this and revisit the Content-Length once we reach a conclusion in the linked issue.

jdm commented:
There are some unit tests that are now failing that ned to be investigated.

tharkum committed:
htmlmediaelement: Support seek requests for non seekable fetch context

According to specification the data:// URL protocol doesn't support range request
so be able make any seek request to required content position let's allow
for non seekable fetch context discard fetched content bytes until seek offset.
https://fetch.spec.whatwg.org/#scheme-fetch

Some scheme URLs (like data:// URL) doesn't expose "Content-Length" header in response
so the total expected size of the stream is unknown and it causes some additional
seek request (SeekData) from the media player. Try to post configure stream size
after we reached fetch EOS response.

Related source code which breaks WPT tests:
tests/wpt/tests/html/canvas/element/manual/imagebitmap/common.sub.js#L56-L78

Testing: Improvements in the following tests:
 - html/canvas/element/manual/imagebitmap/createImageBitmap*
 - mixed-content/gen/top.***/opt-in/video-tag.https.html
 - webgl/tests/conformance/textures/misc/texture-srgb-upload.html

Fixes: servo#32645
Fixes: servo#32745
Fixes: servo#34119
Fixes: servo#34120
Fixes: servo#34151

Signed-off-by: Andrei Volykhin <andrei.volykhin@gmail.com>

tharkum commented:
Change to data:// URL processing ("Content-Length" header) didn't pass unit tests
(expected header maps size should be equal to 1 - "Content-Type" header).

I removed this change  from PR and proposed another change (post configuration of the total stream size) for non seekable fetch context.

tharkum commented:
- Rebased on top of main branch
- Add missing linux WPT results from (https://github.com/tharkum/servo/actions/runs/15488654822/job/43612249050)

tharkum commented:
Linux WPT (**PASSED**): https://github.com/tharkum/servo/actions/runs/15490518178

jdm reviewed (approved):
Nice.
